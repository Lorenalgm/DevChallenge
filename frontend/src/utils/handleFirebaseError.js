export const AuthErrorHandler = {
    generic:
        'Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.',
    'auth/weak-password':
        'Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.',
    'auth/email-already-in-use': 'O email fornecido ja está em uso.',
    'auth/email-already-exists':
        'O e-mail fornecido já está em uso por outro usuário.',
    'auth/internal-error':
        'Erro interno do servidor. Tente novamente mais tarde.',
    'auth/invalid-email': 'Email inválido.',
    'auth/user-not-found':
        'Não há registro de usuário existente correspondente ao identificador fornecido.',
    'auth/wrong-password': 'Senha digitada não esta correta',
    'resource-exhausted':
        'Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.',
};
