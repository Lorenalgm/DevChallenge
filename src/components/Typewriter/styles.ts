import styled from '@emotion/styled'
import { Heading } from '@chakra-ui/react'

export const Typewriter = styled(Heading)`
  border-right: 0.15em solid var(--chakra-colors-secondary-600);
  overflow: hidden;
  white-space: nowrap;
  letter-spacing: 0.02em;
  width: fit-content;
  animation: typing 3.5s steps(30, end), blink-caret 0.5s step-end infinite;

  @keyframes typing {
    from {
      width: 0;
    }
    to {
      width: 100%;
    }
  }

  @keyframes blink-caret {
    from,
    to {
      border-color: transparent;
    }
    50% {
      border-color: var(--chakra-colors-secondary-600);
    }
  }
`
