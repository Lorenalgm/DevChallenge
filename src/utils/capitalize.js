export default function capitalize(word) {
    if (!word) return undefined;

    const firstLetter = word[0];
    const restOfWord = word.slice(1);

    return firstLetter.toUpperCase() + restOfWord;
}
