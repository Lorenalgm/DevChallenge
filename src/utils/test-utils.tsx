import { render, RenderResult } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import theme from 'theme'

export const customRender = (children: React.ReactNode): RenderResult =>
  render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
