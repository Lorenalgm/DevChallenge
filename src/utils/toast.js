import { toast } from 'react-toastify';

const messages = {
  error(msg) {
    return msg || 'Ocorreu um erro no processamento';
  },
  success(msg) {
    return msg || 'Sucesso';
  },
};

const ToastNotification = {
  notify: (method, msg) => {
    const methodStr = method;
    method = toast[method];
    method(messages[methodStr](msg));
  },
};

export default ToastNotification;
