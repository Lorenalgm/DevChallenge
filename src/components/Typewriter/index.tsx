import { Box, HeadingProps } from '@chakra-ui/react'

import * as S from './styles'

export type TypewriterProps = HeadingProps

const Typewriter = ({ ...props }: TypewriterProps) => (
  <Box width="fit-content">
    <S.Typewriter {...props} />
  </Box>
)

export default Typewriter
