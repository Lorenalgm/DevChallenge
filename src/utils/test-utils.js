import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { render } from '@testing-library/react';

import { GlobalStyles } from '../styles/GlobalStyles';

const customRender = (component, { ...renderOptions } = {}) =>
    render(
        <>
            <GlobalStyles />
            {component}
            <ToastContainer />
        </>,
        { wrapper: MemoryRouter, ...renderOptions }
    );

export * from '@testing-library/react';
export { customRender as render };
