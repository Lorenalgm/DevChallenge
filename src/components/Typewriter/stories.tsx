import { Story, Meta } from '@storybook/react/types-6-0'

import Typewriter, { TypewriterProps } from '.'

export default {
  title: 'Typewriter',
  component: Typewriter
} as Meta

export const Default: Story<TypewriterProps> = (args) => (
  <div
    style={{
      paddingTop: '20px',
      display: 'flex',
      width: '100%',
      justifyContent: 'center'
    }}
  >
    <Typewriter {...args} />
  </div>
)

Default.args = {
  children: 'Typewriter'
}
