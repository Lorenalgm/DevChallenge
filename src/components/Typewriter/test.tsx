import { render, screen } from 'utils/test-utils'

import Typewriter from '.'

describe('<Typewriter />', () => {
  it('should render the text passed', () => {
    const { container } = render(<Typewriter>Typewriter Example</Typewriter>)

    expect(
      screen.getByRole('heading', { name: /typewriter example/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
