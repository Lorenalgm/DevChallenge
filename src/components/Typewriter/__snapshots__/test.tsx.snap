// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typewriter /> should render the text passed 1`] = `
.emotion-0 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.emotion-1 {
  font-family: var(--chakra-fonts-heading);
  font-weight: var(--chakra-fontWeights-bold);
  font-size: var(--chakra-fontSizes-3xl);
  line-height: 1.33;
  border-right: 0.15em solid var(--chakra-colors-secondary-600);
  overflow: hidden;
  white-space: nowrap;
  letter-spacing: 0.02em;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  -webkit-animation: typing 3.5s steps(30, end),blink-caret 0.5s step-end infinite;
  animation: typing 3.5s steps(30, end),blink-caret 0.5s step-end infinite;
}

@media screen and (min-width: 48em) {
  .emotion-1 {
    font-size: var(--chakra-fontSizes-4xl);
    line-height: 1.2;
  }
}

<div
  class="emotion-0"
>
  <h2
    class="chakra-heading emotion-1"
  >
    Typewriter Example
  </h2>
</div>
`;
